common/remote/sockfile.cpp

```diff
@@ -6339,6 +6339,8 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
             }
             else
             {
+                responseWriter->outputBeginArray("Msg");
```

Comment:
You should actually begin a nested "Msg" object/item here, as well.  Each group of rows and each Eog is actually going to be inside a Msg object.

 responseWriter->outputBeginNested("Msg");